### PR TITLE
Android

### LINKED ISSUE #19: Android client
I searched but did not find

### PR DESCRIPTION
**Pull request by brunolemos**
Closes #19 

### Required

- [x] Upgrade [JSC](https://github.com/react-community/jsc-android-buildscripts) to fix crashes related to old JavaScript version
- [x] Test everything
- [x] Fix scrollTo not working sometimes after adding a column
- [x] Fix blank screen after removing last column
- [x] Fix Splash Screen
- [x] Fix app not showing on home screen
- [x] Fix release build
- [x] Publish to Google Play Store ([link](https://play.google.com/store/apps/details?id=com.devhubapp))

### Optional
- [x] Animate StatusBar background color transition
- [ ] Use a custom local storage implementation, faster and with increased limit
- [x] On Add Column modal, the FAB should not go up when keyboard is open
- [x] On Add Column modal, close keyboard when tapping outside the input
- [x] On Add Column modal, scroll to input when focusing on it
- [x] Optimize what is saved on cache (remove unused fields from api response, which should improve redux-persist performance and not hit android asyncstorage limit)
- [ ] Sync `Saved for Later` items between multiple devices
- [ ] Open links in app instead of the browser

### CODE REVIEW

**Comment by zhigang1992** [comment]
Also, are you able to do `bundleReleaseJsAndAssets`, it's looking for `devhub/packages/mobile/node_modules/react-native/local-cli/cli.js` and not `devhub/node_modules/react-native/local-cli/cli.js`. But I think it's needed to be able to create a release build?

**Comment by brunolemos** [comment]
@zhigang1992 I have these download errors too, but it works anyway. See stackoverflow: https://stackoverflow.com/questions/53246632/org-gradle-api-resources-resourceexception-could-not-get-resource-in-android-st

> Also, are you able to do bundleReleaseJsAndAssets.

Haven't tried release build yet, but needs to find the place it's looking for `../node_modules/react-native/local-cli/cli.js` and change to `../.././node_modules/react-native/local-cli/cli.js`

**Comment by lucarge** [comment]
Hi @brunolemos, thanks for devhub first of all!

About the issue related to `node_modules/react-native/local-cli/cli.js`, have you considered to symlink `react-native` from the workspace `node_modules` to `mobile/node_modules`?

A lot of react native's internals has hardcoded paths that do not expect to find `react-native` somewhere else 😞 

My suggestion is to symlink it automatically with a postinstall script directly inside the mobile's `package.json` 🙌

**Comment by brunolemos** [comment]
I was able to fix it by setting the `cliPath` on `app/build.gradle`.

```
project.ext.react = [
        bundleAssetName: "index.android.bundle",
        cliPath        : "../../node_modules/react-native/local-cli/cli.js",
        entryFile      : "packages/mobile/index.js"
]
```

**Comment by zhigang1992** [comment]
@brunolemos  Nicely done!
Did you figured it out by going through the source code, `cliPath` is not in the comment above.

@lucarge I think @brunolemos have pretty much gone through all the cases when working on iOS platform. 😄

**Comment by brunolemos** [comment]
I'm publishing an apk right now so you can start testing

File: packages/mobile/android/app/build.gradle [hunk 1]
```diff
@@ -73,8 +73,9 @@ import com.android.build.OutputFile
  */
 
 project.ext.react = [
-    bundleAssetName: "index.android.bundle",
-    entryFile: "index.js"
+        bundleAssetName: "index.android.bundle",
```
Comment (zhigang1992) [inline packages/mobile/android/app/build.gradle]: Is this option necessary, isn't "index.android.bundle" the default value?

**Comment by brunolemos** [comment]
Available now, please test and open issues: https://github.com/devhubapp/devhub/issues/19#issuecomment-448524196

File: packages/mobile/android/app/build.gradle (hunk 1 above)
Comment (brunolemos) [reply packages/mobile/android/app/build.gradle]: I think it's not necessary

**Pull request merged**